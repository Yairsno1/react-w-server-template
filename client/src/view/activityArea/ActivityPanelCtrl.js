import React, {Component} from 'react';
import {routeEnum} from '../../util/navHelper';
import PropTypes from 'prop-types';

function ActivityCrtlMock(props) {
  let c = 'w3-content w3-container w3-padding-16 w3-' + props.color;
  return (
    <div className={c}>
      <div className="w3-round-xlarge" style={{border: 'solid 3px black'}}>
        <div className="w3-container w3-center">
          <h1 dir="ltr" className="w3-text-black">
            {props.text}
          </h1>
        </div>
      </div>
    </div>
  );
}

class ActivityPanelCtrl extends Component {
  render() {
    const classItems = [
      'w3-col',
      's9',
      'm10',
      'l11',
      'w3-padding-48'];

      let activityCtrl = null;

      if (routeEnum.home === this.props.activity) {
        activityCtrl = <ActivityCrtlMock color={"white"} text={"חשבון, זה פשוט"}/>
      } else if (routeEnum.add === this.props.activity) {
        activityCtrl = <ActivityCrtlMock color={"green"} text={"2 + 2"}/>
      } else if (routeEnum.sub === this.props.activity) {
        activityCtrl = <ActivityCrtlMock color={"deep-orange"} text={"6-2"}/>
      } else if (routeEnum.mult === this.props.activity) {
        activityCtrl = <ActivityCrtlMock color={"blue"} text={"1 x 4"}/>
      }  else if (routeEnum.div === this.props.activity) {
        activityCtrl = <ActivityCrtlMock color={"yellow"} text={"8 : 2"}/>
      }

    return (
      <div className={classItems.join(' ')}>
        {activityCtrl}
      </div>
    );
  }
}

ActivityPanelCtrl.propTypes = {
  activity: PropTypes.string.isRequired,
};

export default ActivityPanelCtrl;
