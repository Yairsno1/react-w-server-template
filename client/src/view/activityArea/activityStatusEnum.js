const activityStatusEnum = {
  next: 0,
  q: 1,
  answerOk: 2,
  answerWrong: 3,
  answerShow: 4,
};

export default activityStatusEnum;
