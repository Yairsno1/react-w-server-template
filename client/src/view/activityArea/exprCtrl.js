import React, {Component} from 'react';
//import PropTypes from 'prop-types';

export class QCtrl extends Component {
  render()
  {
    return(
      <div className="w3-container w3-center">
        <p dir="ltr" className="w3-xxxlarge w3-text-black">
          <b>{this.props.text}</b>
        </p>
      </div>
    );
  }
}

export class WaitCtrl extends Component {
  render()
  {
    return(
      <div className="w3-container w3-center">
        <p className="w3-xxxlarge w3-text-black">
          <span className="fa fa-gear w3-xxxlarge w3-spin"></span>
          <span>&#160;</span>
          <b>טוען תרגיל ...</b>
        </p>
      </div>
    );
  }
}
