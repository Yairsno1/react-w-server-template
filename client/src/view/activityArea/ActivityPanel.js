import { connect } from 'react-redux'
import {routeEnum} from '../../util/navHelper';
import ActivityPanelCtrl from './ActivityPanelCtrl';

const mapStateToProps = (state, ownProps) => {
  const activity = ownProps.route ? ownProps.route.toUpperCase() : routeEnum.home;

  return {
    activity: activity,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onNextQ: () => {
      ownProps.onNextQ(dispatch);
    },
    onAnswer: (a) => {
      ownProps.onAnswer(dispatch, a);
    },
  };
};

const ActivityPanel = connect(
  mapStateToProps,
  mapDispatchToProps)(ActivityPanelCtrl);

export default ActivityPanel;
