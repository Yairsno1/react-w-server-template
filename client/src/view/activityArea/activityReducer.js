import activityStatusEnum from './activityStatusEnum';
import {
  GENERATE_Q,
  HAS_A,
  GET_NEXT_Q,
  RETRY_Q,
  SHOW_A} from '../../actions';

export const arInitialState =  {
  status: activityStatusEnum.next,
  qText: ''
};

const activityTextReducer = (
  state=arInitialState, action) => {
  let rv = Object.assign({},state);

  if (GENERATE_Q === action.type || RETRY_Q === action.type) {
    rv = {
      status: action.status,
      qText: action.qText,
    };
  } else if (GET_NEXT_Q === action.type) {
    rv = {
      status: activityStatusEnum.next,
      qText: '',
    };
  } else if (HAS_A === action.type) {
    rv = {
      status: action.status,
      qText: action.qText,
    };
  } else if (SHOW_A === action.type) {
    rv = {
      status: action.status,
      qText: action.qText,
    };
  }

  return rv;
};

export default activityTextReducer;
