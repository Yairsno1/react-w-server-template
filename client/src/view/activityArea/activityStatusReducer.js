import activityStatusEnum from './activityStatusEnum';
import {CHANGE_ACTIVITY_STATUS} from '../../actions';

const activityStatusReducer = (state=activityStatusEnum.next, action) => {
  let rv = state;

  if (CHANGE_ACTIVITY_STATUS === action.type) {
    rv = action.status;
  }

  return rv;
};

export default activityStatusReducer;
