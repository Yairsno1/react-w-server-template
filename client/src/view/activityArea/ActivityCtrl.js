import React, {Component} from 'react';
import activityStatusEnum from './activityStatusEnum';
import {WaitCtrl, QCtrl} from './exprCtrl';
import PropTypes from 'prop-types';

class ActivityCtrl extends Component {

  componentDidMount() {
    this.props.nextQ();
  }

  render() {
    const classItems = [
      'w3-content',
      'w3-container',
      'w3-padding-16',
      'w3-' + this.props.color];

    const style = {
      border: 'solid 3px black'
    };

    let ExprCtrl = null;
    const status = this.props.activityStatus;
    if (activityStatusEnum.next === status) {
      ExprCtrl = <WaitCtrl />;
    } else if (activityStatusEnum.q === status) {
      ExprCtrl = <QCtrl text={this.props.qText}/>;
    }

    return (
      <div className={classItems.join(' ')}>
        <div className="w3-round-xlarge" style={style}>
          {ExprCtrl}
        </div>
      </div>
    );
  }
}

ActivityCtrl.propTypes = {
  color: PropTypes.string.isRequired,
  nextQ: PropTypes.func.isRequired,
};

export default ActivityCtrl;
