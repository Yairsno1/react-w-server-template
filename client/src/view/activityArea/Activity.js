import { connect } from 'react-redux'
import ActivityCtrl from './ActivityCtrl';

const mapStateToProps = (state, ownProps) => {

  return {
    activityStatus: state.activityStatus,
    color: ownProps.color,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    nextQ: () => {
      ownProps.onNextQ(dispatch);
    },
  };
};

const Activity = connect(
  mapStateToProps,
  mapDispatchToProps)(ActivityCtrl);

export default Activity;
