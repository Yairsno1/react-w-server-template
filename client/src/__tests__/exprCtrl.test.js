import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {WaitCtrl, QCtrl} from '../view/activityArea/exprCtrl';

let handleReceiveQ = undefined;

beforeAll ( () => {
  handleReceiveQ = jest.fn();
});

//=====================================================
//  We don't need both smoke test of course, we use both
//just for reference and as an example.

//Smoke test
it('WaitCtrl_rendersWithoutCrashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WaitCtrl receiveQ={handleReceiveQ}/>, div);
});

//Smoke test using Enzyme
it('WaitCtrl_rendersWithoutCrashing (using Enzyme)', () => {
  shallow(<WaitCtrl receiveQ={handleReceiveQ}/>);
});
//=====================================================

//Take a snapshot
it('WaitCtrl_snapshot', () => {
  const waitCtrlComp = renderer.create(<WaitCtrl receiveQ={()=>{}}/>);
  let actualWaitCtrl = waitCtrlComp.toJSON();

  expect(actualWaitCtrl).toMatchSnapshot();
});

it('WaitCtrl_receiveQInitiated', () => {
  shallow(<WaitCtrl receiveQ={handleReceiveQ}/>);

  expect(handleReceiveQ).toHaveBeenCalled();
});

describe('QCtrl', () => {

  it('QCtrl_rendersWithoutCrashing', () => {
    shallow(<QCtrl text={"1 + 1 = ?"}/>);
  });

  it('QCtrl_snapshot', () => {
    const qCtrlComp = renderer.create(<QCtrl text={"1 + 1 = ?"}/>);
    let actualQCtrl = qCtrlComp.toJSON();

    expect(actualQCtrl).toMatchSnapshot();
  });

  it('QCtrl_propText', () => {
    const expectedText = '2 + 2 = ?';
    let actualText = '';

    const qCtrlComp = shallow(<QCtrl text={expectedText}/>);
    const p = qCtrlComp.find('p');

    actualText = p.text();

    expect(actualText).toEqual(expectedText);
  });

});

