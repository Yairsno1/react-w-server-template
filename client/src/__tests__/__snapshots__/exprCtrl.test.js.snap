// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QCtrl QCtrl_snapshot 1`] = `
<div
  className="w3-container w3-center"
>
  <p
    className="w3-xxxlarge w3-text-black"
    dir="ltr"
  >
    <b>
      1 + 1 = ?
    </b>
  </p>
</div>
`;

exports[`WaitCtrl_snapshot 1`] = `
<div
  className="w3-container w3-center"
>
  <p
    className="w3-xxxlarge w3-text-black"
  >
    <span
      className="fa fa-gear w3-xxxlarge w3-spin"
    />
    <span>
       
    </span>
    <b>
      טוען תרגיל ...
    </b>
  </p>
</div>
`;
